lib/Connection.js

```diff
@@ -340,5 +368,34 @@ function prepareHeaders (headers = {}, auth) {
   return headers
 }
 
+function getIssuerCertificate (socket) {
+  let certificate = socket.getPeerCertificate(true)
+  while (certificate && Object.keys(certificate).length > 0) {
+    // invalid certificate
+    if (certificate.issuerCertificate == null) {
+      return null
+    }
+
+    // we have reached the root certificate
+    if (certificate.subject.C === certificate.issuer.C &&
+        certificate.subject.ST === certificate.issuer.ST &&
+        certificate.subject.L === certificate.issuer.L &&
+        certificate.subject.O === certificate.issuer.O &&
+        certificate.subject.OU === certificate.issuer.OU &&
+        certificate.subject.CN === certificate.issuer.CN) {
+      break
+    }
+
+    // For self-signed certificates, `issuerCertificate` may be a circular reference.
```

Comment:
We probably never reach this as by definition a self-signed certificate has an issuer that equals its subject. Practically, the fingerprint256 is the sha256 hash of the contents of the certificate so if the previous `if` clause is not true, there is no way for this `if` clause to be true ( in order for us to reach here, the certificate subject is different than the issuerCertificate subject so the fingerprint can't match) . I say we skip this entirely

Reply:
Good call, it would be even better to just compare the two fingerprints then, which is slightly faster than checking every single property.

Reply:
I think that's fine yes. I'm a bit worried by the nodejs documentation that says "_may be_ a circular reference" because I would expect this this to _always_ be a circular reference for self signed certs and this is what we're checking here, but this worry would be the same for the check we removed, so ++ for the change, let's see if we come about any issues